include/aspect/material_model/compositing.h

```diff
@@ -0,0 +1,147 @@
+       * An enum to define what material properties are available.
+       */
+      enum  MaterialProperty
+      {
+        viscosity,
+        density,
+        thermal_expansion_coefficient,
+        specific_heat,
+        thermal_conductivity,
+        compressibility,
+        entropy_derivative_pressure,
+        entropy_derivative_temperature,
+        reaction_terms
+      };
+
+      static const std::map<std::string, MaterialProperty>
+      property_map =
+      {
+        {"Viscosity", viscosity},
+        {"Density", density},
+        {"Thermal expansion coefficient", thermal_expansion_coefficient},
+        {"Specific heat", specific_heat},
+        {"Thermal conductivity", thermal_conductivity},
+        {"Compressibility", compressibility},
+        {"Entropy derivative pressure", entropy_derivative_pressure},
+        {"Entropy derivative temperature", entropy_derivative_temperature},
+        {"Reaction terms", reaction_terms}
+      };
+    }
+
+    /**
+     * A material model that selects properties from other (non-averaging) material models.
+     * @ingroup MaterialModels
+     */
+
+    template <int dim>
+    class Compositing : public MaterialModel::Interface<dim>, public ::aspect::SimulatorAccess<dim>
+    {
+      public:
+        /**
+         * @copydoc MaterialModel::Interface::evaluate()
+         */
+        virtual
+        void
+        evaluate (const typename Interface<dim>::MaterialModelInputs &in,
+                  typename Interface<dim>::MaterialModelOutputs &out) const;
+
+        /**
+         * @copydoc MaterialModel::Interface::declare_parameters()
+         */
+        static void
+        declare_parameters (ParameterHandler &prm);
+
+        /**
+         * @copydoc MaterialModel::Interface::parse_parameters()
+         */
+        virtual void
+        parse_parameters (ParameterHandler &prm);
+
+        /**
+         * @copydoc MaterialModel::Interface::is_compressible()
+         *
+         * Returns value from material model providing compressibility.
+         */
+        virtual bool is_compressible () const;
+
+        /**
+         * @copydoc MaterialModel::Interface::reference_viscosity()
+         *
+         * Taken from the material model providing viscosities
+         */
+        virtual double reference_viscosity () const;
+
+
+
+      private:
+        /**
+         * Parse a string representing one of the components provided by the material models,
+         * and return the resulting MaterialProperty value.
+         *
+         * @param s String with name of property
+         */
+        Property::MaterialProperty
+        parse_property_name(const std::string &s);
+
+        /**
+         * Copy desired properties from material model,
+         *
+         * @param model_index Internal index for pointer to the material model evaluated
+         * @param base_output Properties generated by the material model specified
+         * @out MaterialModelOutputs to be used.
```

Comment:
From the description I do not quite follow what this function does. Please extend the first sentence a bit. E.g. which material model? Supposedly not this object, but one of the models responsible for one of the properties?

After looking at the parameters closer, I think I got its purpose. Probably it copies all properties that are requested from one particular (sub)-model into the final MaterialModelOutputs object?